csharp/ql/src/semmle/code/csharp/controlflow/ControlFlowGraph.qll

```diff
@@ -761,29 +782,32 @@ module ControlFlow {
           this instanceof SwitchExpr
           or
           this instanceof SwitchCaseExpr
-        }
-      }
-
-      private Expr getObjectCreationArgument(ObjectCreation oc, int i) {
-        i >= 0 and
-        if oc.hasInitializer()
-        then result = getExprChildElement(oc, i + 1)
-        else result = getExprChildElement(oc, i)
-      }
-
-      private class PostOrderElement extends ControlFlowElement {
-        PostOrderElement() {
-          this instanceof StandardExpr or
-          this instanceof JumpStmt or
-          this instanceof ThrowExpr or
-          this instanceof ObjectCreation
+          or
+          this instanceof JumpStmt
+          or
+          this instanceof ThrowExpr
         }
 
         ControlFlowElement getFirstChild() {
-          result = this.(StandardExpr).getFirstChildElement() or
-          result = this.(JumpStmt).getChild(0) or
-          result = this.(ThrowExpr).getExpr() or
-          result = getObjectCreationArgument(this, 0)
+          result = this.(StandardExpr).getFirstChildElement()
```

Comment:
I had the feeling so far that we're building the child/parent relationships in the AST in the order of execution. Why do we need to handle individually these types?

Also, `JumpStmt` feels odd to be mixed with expressions.

Reply:
Again, should be better in the follow-up refactoring.